docker/mongodb-kubernetes-tests/tests/search/search_enterprise_tls.py

```diff
@@ -168,28 +187,128 @@ def test_create_search_resource(mdbs: MongoDBSearch):
 
 
 @mark.e2e_search_enterprise_tls
-def test_wait_for_database_resource_ready(mdb: MongoDB):
-    mdb.assert_abandons_phase(Phase.Running, timeout=300)
+def test_wait_for_mongod_parameters(mdb: MongoDB):
+    # After search CR is deployed, MongoDB controller will pick it up
+    # and start adding searchCoordinator role and search-related
+    # parameters to the automation config.
+    def check_mongod_parameters():
+        parameters_are_set = True
+        pod_parameters = []
+        for idx in range(mdb.get_members()):
+            mongod_config = yaml.safe_load(
+                KubernetesTester.run_command_in_pod_container(
+                    f"{mdb.name}-{idx}", mdb.namespace, ["cat", "/data/automation-mongod.conf"]
+                )
+            )
+            set_parameter = mongod_config.get("setParameter", {})
+            parameters_are_set = parameters_are_set and (
+                "mongotHost" in set_parameter and "searchIndexManagementHostAndPort" in set_parameter
+            )
+            pod_parameters.append(f"pod {idx} setParameter: {set_parameter}")
+
+        return parameters_are_set, f'Not all pods have mongot parameters set:\n{"\n".join(pod_parameters)}'
+
+    run_periodically(check_mongod_parameters, timeout=200)
```

Comment:
I think both this and `test_wait_for_agents_ready` are testing implementation details. I understand that `test_wait_for_agents_ready` is necessary (at least for now, while we are not reporting `Pending`). But do we need to do `test_wait_for_mongod_parameters`?

I'm trying to look at the flow from the user perspective. For me as a user it is important that I can:

1. Deploy `MongoDB`
2. Deploy `MongoDBSearch`
3. Get some indication that everything is ready (we don't have that at the moment unfortunately as you explained in the comment for `test_wait_for_agents_ready`).
4. Create search index
5. And start searching.

Can we just drop `test_wait_for_mongod_parameters` and verify "I can search" functionality (you already do that later in the test)?